src/ru/itmo/kotlin/plugin/fir/checkers/PluginErrors.kt

```diff
@@ -0,0 +1,8 @@
+package ru.itmo.kotlin.plugin.fir.checkers
+
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.diagnostics.error0
+
+object PluginErrors {
+    val ILLEGAL_ANNOTATED_FUNCTION by error0<PsiElement>()
```

Comment:
It's worth to add positioning strategy to diagnostic, so compiler will report it only on name function instead of the whole declaration
```kotlin
... by error0<PsiElement>(SourceElementPositioningStrategies.NAME_IDENTIFIER)
```